Migration guide for deprecations removed after v3.13 (#9394)
flutter/website: src/release/breaking-changes/3-13-deprecations.md

---
title: Deprecated API removed after v3.13

> description: >
> After reaching end of life, the following deprecated APIs
> were removed from Flutter.

Reviewer: **chore:** The [build error](https://github.com/flutter/website/actions/runs/6190333187/job/16806357669?pr=9394#step:3:1136) exists because this YAML value needs to be indented.
```suggestion
description: >
  After reaching end of life, the following deprecated APIs
  were removed from Flutter.
```

---

## Summary

In accordance with Flutter's [Deprecation Policy][],
deprecated APIs that reached end of life after the
3.13 stable release have been removed.

All affected APIs have been compiled into this
primary source to aid in migration. A
[quick reference sheet][] is available as well.

[Deprecation Policy]: {{site.repo.flutter}}/wiki/Tree-hygiene#deprecation
[quick reference sheet]: {{site.url}}/go/deprecations-removed-after-3-13

## Changes

This section lists the deprecations, listed by the package and affected class.

### Chip classes' useDeleteButtonTooltip

Package: flutter
Supported by Flutter Fix: yes

The `useDeleteButtonTooltip` property of the following classes was deprecated
in v2.10:

* `DeletableChipAttributes`
* `Chip`
* `RawChip`
* `InputChip`

> The replacement is the `deleteButtonTooltipMessage`. This change simplified the

Reviewer 2: this not follow this: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#use-correct-grammar 

or are there exception in this case, @atsansone ?

API, as providing an empty String to `deleteButtonTooltipMessage` achieves the
same result as setting the original property `useDeleteButtonTooltip` to false.
When `deleteButtonTooltipMessage` is unset, the
`MaterialLocalizations.deleteButtonTooltip` is used by default.

This update to chips & tooltips are covered more extensively in
the [Deprecate useDeleteButtonTooltip for Chips][] design document. An
[in-depth migration guide is also available][].

[Deprecate useDeleteButtonTooltip for Chips]: https://docs.google.com/document/d/1wc9ot7T2E7hJubYxEWMX230a79wYSiFey4BHxnEzHtw/edit?usp=sharing&resourcekey=0-Bo7KPqEtkWgZcSuRCqwQ5w
[in-depth migration guide is also available]: {{site.url}}/release/breaking-changes/chip-usedeletebuttontooltip-migration

> **Migration guide**

Author: This formatting is the same in all past migration guides, so I am going to leave it as is.

The `instantiateImageCodec` method of `PaintingBinding`, as well as the `load`
method of `ImageProvider` and the associated `DecoderCallback` were all
deprecated in v2.13.

The replacement respective replacements are
`PaintingBinding.instantiateImageCodecFromBuffer`, `ImageProvider.loadBuffer`
and `DecoderBufferCallback`.

This change enabled faster performance in image loading by using a buffer.

**Migration guide**

Code before migration:

```dart
PaintingBinding.instance.instantiateImageCodec
```

Code after migration:

```dart
PaintingBinding.instance.instantiateImageCodecFromBuffer
```

**References**

API documentation:

* [`PaintingBinding`][]
* [`ImageProvider`][]
* [`DecoderBufferCallback`][]

Relevant PRs:

* Deprecated in [#103496][]
* Removed in [#132679][]

[`PaintingBinding`]: {{site.api}}/flutter/painting/PaintingBinding-mixin.html
[`ImageProvider`]: {{site.api}}/flutter/painting/ImageProvider-class.html
[`DecoderBufferCallback`]: {{site.api}}/flutter/painting/DecoderBufferCallback.html

[#103496]: {{site.repo.flutter}}/pull/103496
[#132679]: {{site.repo.flutter}}/pull/132679

---

### TestWindow properties

Package: flutter_test
Supported by Flutter Fix: no

> In preparation for multi-window support, many deprecated properties from
> `TestWindow`, which itself is entirely deprecated, has been removed. While the
> deprecation timeframe on `TestWindow` has not yet elapsed, migrating the
> expired properties this cycle will be excellent preparation for fully migrating
> off of `TestWindow`.

Reviewer: **issue (blocking):** This reads with some difficulty.
```suggestion
To prepare for multi-window support,
many deprecated properties of `TestWindow` have been removed.
While `TestWindow` has been deprecated, it does not qualify
for removal at this time.
Migrating the expired properties now will help in migrating from `TestWindow`.
```

The following properties were removed:

* `localeTestValue`
* `clearLocaleTestValue`
* `localesTestValue`
* `clearLocalesTestValue`
* `initialLifecycleStateTestValue`
* `textScaleFactorTestValue`
* `clearTextScaleFactorTestValue`
* `platformBrightnessTestValue`
* `clearPlatformBrightnessTestValue`
* `alwaysUse24HourFormatTestValue`
* `clearAlwaysUse24HourTestValue`
* `brieflyShowPasswordTestValue`
* `defaultRouteNameTestValue`
* `clearDefaultRouteNameTestValue`
* `semanticsEnabledTestValue`
* `clearSemanticsEnabledTestValue`
* `accessibilityFeaturesTestValue`
* `clearAccessibilityFeaturesTestValue`

> This update to `TestWindow` is covered more extensively in the
> [in-depth migration guide][].
> 
> [in-depth migration guide is also available]: {{site.url}}/release/breaking-changes/window-singleton

Reviewer: **suggestion (non-blocking):** Consider a more concise phrasing. (Also update the link to match.)
```suggestion
To learn more about this `TestWindow` update, check out
[`TestWindow` migration guide][].

[`TestWindow` migration guide]: {{site.url}}/release/breaking-changes/window-singleton
```
